### DISCUSSION TITLE
Highlights inside of a textarea

### DISCUSSION THREAD

**Discussion by RiyanParvez** [zauberzeug/nicegui#1218] [Ideas / Feature Requests]
### Description

Is it possible to get highlights inside the ui.textarea like this: 
![image](https://github.com/zauberzeug/nicegui/assets/127128626/414bcc31-0bfb-46ff-88d8-05cc14298b6c)

this is the website by the way: https://nat.dev/
if you get a prompt saying that you need to pay just click on the close icon then out of the models choose LLaMA 2 70b or something since its free, then you can see this in real life. Maybe usefull if you wish to check it with inspect.

**Comment by falkoschindler** [comment]
Hi @RiyanParvez,

Here is a rather hacky solution:

```py
async def select(textarea: ui.textarea, substring: str) -> None:
    start = (textarea.value or '').find(substring)
    await ui.run_javascript(f'''
        const element = getElement({textarea.id}).$refs.qRef.getNativeElement();
        element.focus();
        element.setSelectionRange({start}, {start + len(substring)});
    ''', respond=False)

textarea = ui.textarea(value='Lorem ipsum dolor sit amet, consectetur adipiscing elit.')
ui.button('Select"ipsum"', on_click=lambda: select(textarea, 'ipsum'))
```

I'll convert this discussion into a feature request so we can vote on it and track its progress.

**Reply by RiyanParvez** [reply]
Hello! I tried that with this code:

```py
from nicegui import ui
async def select(textarea: ui.textarea, substring: str) -> None:
    start = (textarea.value or '').find(substring)
    print(start)
    await ui.run_javascript(f'''
        const element = getElement({textarea.id}).$refs.qRef.getNativeElement();
        element.focus();
        element.setSelectionRange({start}, {start + len(substring)});
    ''', respond=False)

textarea = ui.textarea(value='Lorem ipsum dolor sit amet, consectetur adipiscing elit.')
ui.button('Select "ipsum"', on_click=lambda: select(textarea, 'ipsum'))
ui.run()
```

but it doesn't seem to be working, its not doing anything when i click on the button.

**Reply by RiyanParvez** [reply]
And, correct me if i am wrong but to select only the substring shoul'nt it be  {start + len(substring) - 1}?

**Reply by falkoschindler** [reply]
Well, for me it's working exactly as it should (for Chrome, Firefox and Safari). Do you have the latest NiceGUI version installed?

**Reply by RiyanParvez** [reply]
Uhm, it now works after i updated nicegui, but this is not what i was looking for. please check the site i linked earlier once

**Reply by falkoschindler** [reply]
Sorry, I won't sign up for a service just to find out what you're trying to do.
But if you want to add background color to individual lines of text in a textarea, I fear this is impossible without very advanced trickery.

**Comment by rodja** [comment]
Maybe this could be achievable with #1124?

**Reply by falkoschindler** [reply]
I think you better stack different labels on top of an input element. Or use our chat message elements.

**Comment by falkoschindler** [comment]
What I've seen so far suggests that the window you're referring to is not a textarea.

What about
```py
with ui.element().classes('w-60 h-32 p-4 shadow-inner overflow-auto'):
    ui.label('Hello, Whats your name?')
    ui.label("Hi there! My name is LLaMA, I'm a large language model trained by a team of researcher at Meta Al. What can I help you with today?") \
        .classes('bg-gray-200')
```

or even
```py
with ui.element().classes('w-60 h-32 p-4 shadow-inner overflow-auto'):
    ui.label('Hello, Whats your name?').classes('inline')
    ui.label("Hi there! My name is LLaMA, I'm a large language model trained by a team of researcher at Meta Al. What can I help you with today?") \
        .classes('bg-gray-200 inline')
```

<img width="512" alt="Screenshot 2023-07-22 at 07 45 43" src="https://github.com/zauberzeug/nicegui/assets/5767091/948e0f8b-701c-4b05-9995-47c3e579dc92">

**Reply by RiyanParvez** [reply]
this can'nt be modifyied (nothing can be written or removed ) like a textarea while the element inside the pic i showed can be.

**Reply by RiyanParvez** [reply]
![image](https://github.com/zauberzeug/nicegui/assets/127128626/ac4630a2-6cf6-4e79-9b68-2d5f90bc44ac)

**Comment by WolfgangFahl** [comment]
I am not doing things with the textarea directly but switch between a html view and the textarea with a click of a button. Im my case im am doing syntax highlighting see http://nicescad.bitplan.com/
